packages/host/app/lib/matrix-handlers/membership.ts

```diff
@@ -69,32 +83,34 @@ export function onMembership(context: Context) {
         });
       }
       if (member.membership === 'join') {
-        context.roomMembershipQueue.push({
+        myMemberships.push({
           type: 'join',
           roomId,
           eventId,
           timestamp,
         });
       }
       if (member.membership === 'leave') {
-        context.roomMembershipQueue.push({ type: 'leave', roomId });
+        myMemberships.push({ type: 'leave', roomId });
       }
-      flushMembershipQueue(context);
     }
-  };
+  }
+
+  processMyMemberships(context, myMemberships);
```

Comment:
I bet we can get rid of this and use the room card state to figure out all the room's you have been invited to or joined. I'll refactor this out